tests/test-hostinfo.lua

```diff
@@ -36,4 +42,126 @@ require('tap')(function(test)
+      inFixture:write(chunk)
+      inFixture:write(nil)
+    end)
+  end
+
+  ---------------------------------------------- Tests for Autoupdates ------------------------------------------------
+
+  test('test for Autoupdates: apt reader enabled', function()
+    local autoupdates = require('../hostinfo/autoupdates')
+    local errTable, outTable = {}, {}
+    local reader = autoupdates.AptReader:new()
+    local inFixture = LineEmitter:new()
+    local outFixture = hostinfoFixtureDir['autoupdates_apt_out.txt']
+    inFixture:pipe(reader)
+    reader:on('error', function(err) table.insert(errTable, err) end)
+    reader:on('data', function(data) table.insert(outTable, data) end)
+    reader:once('end', function()
+      local outFixtureTable = json.parse(outFixture)
+      assert(is_table_equal(outFixtureTable, outTable), 'not ok: outFixture and outTable dont match')
+      assert(#errTable==0, 'Not ok: errtable is not 0 length')
+    end)
+    local chunk = hostinfoFixtureDir['autoupdates_apt_in.txt']
+    inFixture:write(chunk)
+    inFixture:write(nil)
+  end)
+
+  test('test for Autoupdates: yum reader enabled', function()
+    local autoupdates = require('../hostinfo/autoupdates')
+    local errTable, outTable = {}, {}
+    local reader = autoupdates.YumReader:new()
+    local inFixture = LineEmitter:new()
+    local outFixture = hostinfoFixtureDir['autoupdates_yum_out.txt']
+    inFixture:pipe(reader)
+    reader:on('error', function(err) table.insert(errTable, err) end)
+    reader:on('data', function(data) table.insert(outTable, data) end)
+    reader:once('end', function()
+      local outFixtureTable = json.parse(outFixture)
+      assert(is_table_equal(outFixtureTable, outTable), 'not ok: outFixture and outTable dont match')
+      assert(#errTable==0, 'Not ok: errtable is not 0 length')
+    end)
+    local chunk = hostinfoFixtureDir['autoupdates_yum_in.txt']
+    inFixture:write(chunk)
+    inFixture:write(nil)
+  end)
+
+  ----------------------------------------------- Tests for packages --------------------------------------------------
+
+  test('test for packages: linux reader', function()
+    local autoupdates = require('../hostinfo/packages')
+    local errTable, outTable = {}, {}
+    local reader = autoupdates.LinuxReader:new()
+    local inFixture = LineEmitter:new()
+    local outFixture = hostinfoFixtureDir['packages_linux_out.txt']
+    inFixture:pipe(reader)
+    reader:on('error', function(err) table.insert(errTable, err) end)
+    reader:on('data', function(data) table.insert(outTable, data) end)
+    reader:once('end', function()
+      local outFixtureTable = json.parse(outFixture)
+      assert(is_table_equal(outFixtureTable, outTable), 'not ok: outFixture and outTable dont match')
+      assert(#errTable==0, 'Not ok: errtable is not 0 length')
+    end)
+    local chunk = hostinfoFixtureDir['packages_linux_in.txt']
+    inFixture:write(chunk)
+    inFixture:write(nil)
+  end)
```

Comment:
nice job... as I explained in IRC, let's tweak these test functions to use 'expect' on some of the callbacks, namely the ones for 'end'.